src/components/error/error.js

```diff
@@ -0,0 +1,51 @@
+import Events from '../../base/events'
+import BaseObject from '../../base/base_object'
+import Log from '../../plugins/log'
+
+/**
+ * The PlayerError is responsible to receive and propagate errors.
+ * @class PlayerError
+ * @constructor
+ * @extends BaseObject
+ * @module components
+ */
+class PlayerError extends BaseObject {
+  get name() { return 'error' }
+
+  /**
+   * @property Levels
+   * @type {Object} object with error levels
+   */
+  get Levels() {
+    return {
+      FATAL: 'FATAL',
+      WARN: 'WARN'
+    }
+  }
+
+  constructor(options={}) {
+    super(options)
+  }
+
+  /**
+   * creates and trigger an error.
+   * @method error
+   * @param {Object} err should be an object with code, description, level, origin, scope and raw error.
+   */
+  error(err) {
+    if (!this.core) {
+      Log.warn(this.name, 'Core is not setted. Error: ', err)
+      return
+    }
+    this.core.trigger(Events.ERROR, err)
+  }
+
+  setCore(core) {
+    this.core = core
+  }
+
+}
+
+const playerError = new PlayerError()
```

Comment:
I wouldn't recommend using a singleton approach for this component. Since there could be more than a single player in a given context, an error would always propagate to the last instance which called `setCore`.

Reply:
Instead of doing that, we could create the error component when the core is instantiated, perhaps in the `CoreFactory` or in the `Core` itself.

Reply:
fixed on cb5fe1fbce139bca7d01f5c48c8d6ba63e2cae31